packages/ra-core/src/controller/ListController.tsx

```diff
@@ -207,24 +206,14 @@ export class UnconnectedListController extends Component<
 
     componentWillReceiveProps(nextProps: Props & EnhancedProps) {
         if (
+            nextProps.version !== this.props.version ||
             nextProps.resource !== this.props.resource ||
-            nextProps.query.sort !== this.props.query.sort ||
-            nextProps.query.order !== this.props.query.order ||
-            nextProps.query.page !== this.props.query.page ||
-            nextProps.query.perPage !== this.props.query.perPage ||
-            !isEqual(nextProps.query.filter, this.props.query.filter) ||
+            !isEqual(nextProps.query, this.props.query) ||
             !isEqual(nextProps.filter, this.props.filter) ||
             !isEqual(nextProps.sort, this.props.sort) ||
             !isEqual(nextProps.perPage, this.props.perPage)
         ) {
-            this.updateData(
-                Object.keys(nextProps.query).length > 0
```

Comment:
nextProps.query isn't the same as the result of `getQuery`, which does the fallback to params. So I think you still need the params fallback here.

Reply:
You mean we should check if `nextProps.params` has changed ? Because we are now passing the props to `updateData`